go.mod

```diff
@@ -1,15 +1,19 @@
 module github.com/Scalingo/etcd-discovery/v7
 
-go 1.15
+go 1.16
 
 require (
-	github.com/coreos/go-semver v0.3.0 // indirect
 	github.com/gofrs/uuid v3.4.0+incompatible
 	github.com/golang/mock v1.6.0
-	github.com/json-iterator/go v1.1.10 // indirect
 	github.com/smartystreets/goconvey v1.6.4
-	go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200826232710-c20cc05fc548
-	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9 // indirect
-	golang.org/x/net v0.0.0-20210405180319-a5a99cb37ef4
+	go.etcd.io/etcd/client/pkg/v3 v3.5.0
+	// The latest versions of etcd have been migrated to go modules.
+	// Since this change the version of the etcd client we are currently
+	// using in this package has been named v2.
+	// This does not mean that it does not work with the etcd server version 3.
+	//
+	// The package "go.etcd.io/etcd/client/v3" is a complete refactoring
+	// of the client and uses grpc instead of http, that we currently use.
```

Comment:
I imagine that migrating this project to gRPC is a big task, that's why you don't do it?

Reply:
Many functions, methods, const changes yes. This should be done in another PR, it's too big for now.

Reply:
I have opened an issue: https://github.com/Scalingo/etcd-discovery/issues/50